crates/core/tedge_mapper/Cargo.toml

```diff
@@ -33,9 +34,9 @@ c8y_translator = { path = "../c8y_translator" }
 chrono = "0.4"
 clock = { path = "../../common/clock" }
 csv = "1.1"
+download = { path = "../../common/download" }
```

Comment:
I'm surprised to see `download` as a *new* dependency, since the mapper is actually doing no download.

* We have a `pub use download::*;` in the `agent_interface`. That should be enough to manipulate the download infos.

Reply:
Mapper uses the REST APIs to send the software list operation result. The URL creation and authorization for REST APIs uses some code from the download crate, that's the reason why the mapper still needs the download crate. The code related to the URL can be moved to a new URL crate.